manager/dispatcher/dispatcher_test.go

```diff
@@ -714,22 +825,37 @@ func TestAssignmentsAddingTasks(t *testing.T) {
 // If a secret or config is updated or deleted, even if it's for an existing task, no changes will be sent down
 func TestAssignmentsDependencyUpdateAndDeletion(t *testing.T) {
 	t.Parallel()
+	testFuncs := []taskGeneratorFunc{makeTasksAndDependenciesWithResourceReferences, makeTasksAndDependenciesNoResourceReferences, makeTasksAndDependenciesOnlyResourceReferences, makeTasksAndDependenciesWithRedundantReferences}
+	for _, testFunc := range testFuncs {
+		testAssignmentsDependencyUpdateAndDeletionWithGivenTasks(t, testFunc)
+	}
+}
 
+func testAssignmentsDependencyUpdateAndDeletionWithGivenTasks(t *testing.T, genTasks taskGeneratorFunc) {
 	gd, err := startDispatcher(DefaultConfig())
 	assert.NoError(t, err)
 	defer gd.Close()
 
 	expectedSessionID, nodeID := getSessionAndNodeID(t, gd.Clients[0])
 
 	// create the relevant secrets and tasks
-	secrets, configs, tasks := makeTasksAndDependencies(t, nodeID)
+	secrets, configs, resourceRefs, tasks := genTasks(t, nodeID)
 	err = gd.Store.Update(func(tx store.Tx) error {
 		for _, secret := range secrets {
-			assert.NoError(t, store.CreateSecret(tx, secret))
+			if store.GetSecret(tx, secret.GetID()) == nil {
+				assert.NoError(t, store.CreateSecret(tx, secret))
+			}
 		}
 		for _, config := range configs {
-			assert.NoError(t, store.CreateConfig(tx, config))
+			if store.GetConfig(tx, config.GetID()) == nil {
```

Comment:
nit: Just use `secret.ID`. `GetID` exists for type-agnostic code that uses these through an interface.